wrappers/desktop.go

```diff
@@ -102,32 +102,72 @@ var isValidDesktopFileLine = regexp.MustCompile(strings.Join([]string{
 	"^TargetEnvironment=",
 }, "|")).Match
 
-// rewriteExecLine rewrites a "Exec=" line to use the wrapper path for snap application.
-func rewriteExecLine(s *snap.Info, desktopFile, line string) (string, error) {
+type execRewriterImpl struct {
+	// strict enables fallback to matching a desktop file with snap
+	// application using the application's name
+	strict bool
+	// rewriter matches given app with a desktop file and returns a rewritten
+	// exec line or an empty string
+	rewriter func(app *snap.AppInfo, desktopFile, execLine string) string
```

Comment:
There's quite a bit of complexity here which feels undue given the problem at hand: a new interface, a new type, that has an actual field that is in fact the only method in that interface, two different implementations that match not the interface but the method, test functions that create an interface by using the methods on the type...

Yet, all of that in the name of a very simple change in a simple algorithm. I may be missing something, but feels unnecessary.

Reply:
The complexity comes is a result of reviews of the previous version of the PR #4768. The actual change is very simple and I'm ok with keeping it in `userd` package or at least closer to the autostart code just like it was done in the previous 2 iterations of this functionality.